packages/cli/lib/base-generator.js

```diff
@@ -28,11 +32,235 @@ module.exports = class BaseGenerator extends Generator {
+      def = await question.default(answers);
+    }
+    let defaultVal = def;
+
+    if (def == null) {
+      // No `default` is set for the question, check existing answers
+      defaultVal = answers[question.name];
+      if (defaultVal != null) return defaultVal;
+    }
+
+    if (question.type === 'confirm') {
+      return defaultVal != null ? defaultVal : true;
+    }
+    if (question.type === 'list' || question.type === 'rawList') {
+      // Default to 1st item
+      if (def == null) def = 0;
+      if (typeof def === 'number') {
+        // The `default` is an index
+        const choice = question.choices[def];
+        if (choice) {
+          defaultVal = choice.value || choice.name;
+        }
+      } else {
+        // The default is a value
+        if (question.choices.map(c => c.value || c.name).includes(def)) {
+          defaultVal = def;
+        }
+      }
+    } else if (question.type === 'checkbox') {
+      if (def == null) {
+        defaultVal = question.choices
+          .filter(c => c.checked && !c.disabled)
+          .map(c => c.value || c.name);
+      } else {
+        defaultVal = def
+          .map(d => {
+            if (typeof d === 'number') {
+              const choice = question.choices[d];
+              if (choice && !choice.disabled) {
+                return choice.value || choice.name;
+              }
+            } else {
+              if (
+                question.choices.find(
+                  c => !c.disabled && d === (c.value || c.name),
+                )
+              ) {
+                return d;
+              }
+            }
+            return undefined;
+          })
+          .filter(v => v != null);
+      }
+    }
+    return defaultVal;
+  }
+
+  /**
+   * Override the base prompt to skip prompts with default answers
+   * @param questions One or more questions
+   */
+  async prompt(questions) {
+    // Normalize the questions to be an array
+    if (!Array.isArray(questions)) {
+      questions = [questions];
+    }
+    if (!this.options['yes']) {
+      if (!process.stdin.isTTY) {
+        this.log(
+          chalk.red('The stdin is not a terminal. No prompt is allowed.'),
+        );
+        process.exit(1);
+      }
+      // Non-express mode, continue to prompt
+      return await super.prompt(questions);
+    }
+
+    const answers = Object.assign({}, this.options);
+
+    for (const q of questions) {
+      let when = q.when;
+      if (typeof when === 'function') {
+        when = await q.when(answers);
+      }
+      if (when === false) continue;
+      if (this._isQuestionOptional(q)) {
+        const answer = await this._getDefaultAnswer(q, answers);
+        debug('%s: %j', q.name, answer);
+        answers[q.name] = answer;
+      } else {
+        if (!process.stdin.isTTY) {
+          this.log(
+            chalk.red('The stdin is not a terminal. No prompt is allowed.'),
+          );
+          process.exit(1);
```

Comment:
Ditto. Don't exit the process, end the generator execution only.

Reply:
Fixed